dhall/benchmark/parser/Main.hs

```diff
@@ -67,27 +68,25 @@ benchExprFromText name expr =
 
 benchExprFromBytes
     :: String -> Data.ByteString.Lazy.ByteString -> Gauge.Benchmark
-benchExprFromBytes name bytes = bench name (whnf f bytes)
+benchExprFromBytes name bytes = bench name (nf f bytes)
   where
     f bytes = do
-        term <- case Codec.Serialise.deserialiseOrFail bytes of
-            Left  _    -> Nothing
-            Right term -> return term
-        case Dhall.Binary.decodeExpression term
-          :: Either Dhall.Binary.DecodingFailure (Dhall.Expr () Dhall.Import) of
+        case Dhall.Binary.decodeExpression bytes of
             Left  _          -> Nothing
```

Comment:
This is unrelated to your changes, but I think we should make this an error, so we'll know when decoding fails.